src/Illuminate/Database/Concerns/ManagesTransactions.php

```diff
@@ -194,12 +194,12 @@ public function commit()
             $this->getPdo()->commit();
         }
 
-        $this->transactions = max(0, $this->transactions - 1);
-
         if ($this->afterCommitCallbacksShouldBeExecuted()) {
             $this->transactionsManager?->commit($this->getName());
         }
 
+        $this->transactions = max(0, $this->transactions - 1);
```

Comment:
Reset `$this->transactions` after calling `afterCommitCallbacksShouldBeExecuted()` so we can get the true transaction levels during the commit.